src/store/RelayQueryWriter.js

```diff
@@ -551,6 +551,9 @@ class RelayQueryWriter extends RelayQueryVisitor<WriterState> {
 
     this._store.putLinkedRecordIDs(recordID, storageKey, nextLinkedIDs);
 
+    // Check if length has changed
+    isUpdate = isUpdate || (prevLinkedIDs && prevLinkedIDs.length || 0) !== nextLinkedIDs.length;
```

Comment:
Sorry, one more super small nit: can you reformat this to fit in than 80 chars?
```
isUpdate = isUpdate ||
  (prevLinkedIDs && prevLinkedIDs.length || 0) !== nextLinkedIDs.length;
```

I'll merge it after that tweak.